presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -2142,6 +2203,30 @@ else if (column.isHidden()) {
         return columnHandles.build();
     }
 
+    private static void validateCsvColumns(ConnectorTableMetadata tableMetadata)
+    {
+        if (getHiveStorageFormat(tableMetadata.getProperties()) != HiveStorageFormat.CSV) {
+            return;
+        }
+
+        List<String> partitionedBy = getPartitionedBy(tableMetadata.getProperties());
```

Comment:
Not that it really matters, but make this a `Set` so that contains is more efficient